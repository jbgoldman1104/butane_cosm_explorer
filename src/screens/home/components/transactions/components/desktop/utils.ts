export const columns:{
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number,
}[] = [
  {
    key: 'block',
    width: 25,
  },
  {
    key: 'hash',
    width: 25,
  },
  {
    key: 'result',
    align: 'right',
    width: 25,
  },
  {
    key: 'time',
    align: 'right',
    width: 25,
  },
];
