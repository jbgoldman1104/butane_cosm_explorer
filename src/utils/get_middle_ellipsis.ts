/**
 * Helper function to help generate ellpsis in the middle of text
*/
export const getMiddleEllipsis = (str:string, options?: {
  beginning?: number,
  ending?: number,
}) => {
  const oneThird = Math.floor(str.length / 3);
  const {
    beginning = oneThird + oneThird,
    ending = oneThird,
  } = options ?? {
  };

  const startEndTotal = beginning + ending;

  if (startEndTotal && startEndTotal < str.length) {
    return `${str.substr(0, beginning)}...${str.substr(str.length - ending, str.length)}`;
  }

  return str;
};
