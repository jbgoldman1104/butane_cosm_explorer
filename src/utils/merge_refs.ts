/* eslint-disable no-restricted-syntax */

/**
 * helper function to merge multiple refs
 * @param refs
 */
export const mergeRefs = (...refs) => {
  const filteredRefs = refs.filter(Boolean);
  if (!filteredRefs.length) return null;
  if (filteredRefs.length === 1) return filteredRefs[0];
  return (inst) => {
    for (const ref of filteredRefs) {
      if (typeof ref === 'function') {
        ref(inst);
      } else if (ref) {
        ref.current = inst;
      }
    }
  };
};
