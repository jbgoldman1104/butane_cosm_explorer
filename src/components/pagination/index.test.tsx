import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import Pagination from '.';

// ==================================
// mocks
// ==================================
jest.mock('./components', () => ({
  Actions: (props) => <div id="Actions" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('components: Pagination', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <Pagination
          total={100}
          handleChangePage={jest.fn()}
          handleChangeRowsPerPage={jest.fn()}
          page={0}
          rowsPerPage={5}
        />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
