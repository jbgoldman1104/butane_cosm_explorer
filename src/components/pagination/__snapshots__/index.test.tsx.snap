// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: Pagination matches snapshot 1`] = `
<div
  className="MuiTablePagination-root makeStyles-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
  >
    <div
      className="MuiTablePagination-spacer"
    />
    <p
      className="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
    >
      common:paginationLabelOne
    </p>
    <div
      backIconButtonProps={
        Object {
          "aria-label": "Previous page",
          "title": "Previous page",
        }
      }
      className="makeStyles-mobile"
      count={100}
      handleChangeRowsPerPage={[MockFunction]}
      id="Actions"
      nextIconButtonProps={
        Object {
          "aria-label": "Next page",
          "title": "Next page",
        }
      }
      onChangePage={[MockFunction]}
      page={0}
      rowsPerPage={5}
    />
    <div
      backIconButtonProps={
        Object {
          "aria-label": "Previous page",
          "title": "Previous page",
        }
      }
      className="makeStyles-tablet"
      count={100}
      handleChangeRowsPerPage={[MockFunction]}
      id="Actions"
      nextIconButtonProps={
        Object {
          "aria-label": "Next page",
          "title": "Next page",
        }
      }
      onChangePage={[MockFunction]}
      page={0}
      pageNeighbors={2}
      rowsPerPage={5}
    />
  </div>
</div>
`;
